### PR TITLE
Add split offset function v0.6

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #2307 

---

# Add helper function SplitOffset needed for shuffling

includes:
TestSplitOffset_OK
SplitOffset

**Write why you are making the changes in this pull request**
to comply with 0.6 spec

**Write a summary of the changes you are making**
changes in:
- beacon-chain/utils/shuffle_test.go
- beacon-chain/utils/shuffle.go
**Link anything that would be helpful or relevant to the reviewers**
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#get_split_offset

### CODE REVIEW

**Comment by shayzluf** [comment]
there is no splitoffset function

**Comment by terencechain** [comment]
> there is no splitoffset function @terencechain

It's a split function. `SplitIndices` essentially does the same thing and we only need one, so I'd say delete that and use your `SplitOffset` or update the name of `SplitIndices` to `SplitOffset`. We are not just implementing the new functions we have to replace/fix current ones too : )

**Comment by shayzluf** [comment]
you are right, i thought it was an added function. didn't notice that it changed the old function. its an improvement to the other one. fixing
Thanks

File: beacon-chain/utils/shuffle_test.go [hunk 1]
```diff
@@ -72,3 +72,20 @@ func TestSplitIndices_OK(t *testing.T) {
...
+	var l []uint64
+	validators := uint64(64000)
+	for i := uint64(0); i < validators; i++ {
+		l = append(l, i)
+	}
+	chunks := uint64(6)
+	split := SplitIndices(l, chunks)
+	for i := uint64(0); i < chunks; i++ {
+		if !reflect.DeepEqual(split[i], l[SplitOffset(uint64(len(l)), chunks, i):SplitOffset(uint64(len(l)), chunks, i+1)]) {
+			t.Errorf("SplitOffset doesnt function as intended. want: %v got: %v", l[SplitOffset(uint64(len(l)), chunks, i):SplitOffset(uint64(len(l)), chunks, i+1)], split[i])
```
Comment (terencechain) [inline beacon-chain/utils/shuffle_test.go]:
```suggestion
			t.Errorf("Want: %v got: %v", l[SplitOffset(uint64(len(l)), chunks, i):SplitOffset(uint64(len(l)), chunks, i+1)], split[i])
```

File: beacon-chain/utils/shuffle_test.go [hunk 2]
```diff
@@ -72,3 +72,20 @@ func TestSplitIndices_OK(t *testing.T) {
 		}
 	}
 }
+
+func TestSplitOffset_OK(t *testing.T) {
```
Comment (prestonvanloon) [inline beacon-chain/utils/shuffle_test.go]: Can you add a test that tests splitoffset only? This test seems to depend on SplitIndicies

Comment (nisdas) [reply beacon-chain/utils/shuffle_test.go]: addressed here https://github.com/prysmaticlabs/prysm/pull/2365/commits/d0e4a4772ab0fffa2d7e632e2cad9cf0f3030af3

**Pull request merged**